### ISSUE TITLE
[tbendfelt on 2016-12-20 11:28:01] : pDAQ: StringHub : Dial down the amount of buffering in the hub-to-trigger hit channel.

### ISSUE THREAD

**Issue by dglo** [dglo/config#115]
[main 75ac49b] 1 changed, 1 inserted, 1 deleted

**Comment by dglo** [comment]
[tbendfelt on 2016-12-20 12:08:01]
NOTE:  There is a twist to the watchdog that makes this difficult to reproduce on SPTS.  On SPS the watchdog period is 10 seconds with a limit of 10 periods with a stagnant component.  A quick analysis might conclude that even at 30K/sec, data should be released ever 5 seconds and the watchdog would never fail. However, the watchdog is tracking multiple counters each of which is affected by the buffer at a different rate.  If the periods of these counters line up such that 10 consecutive watchdog intervals find a stagnant counter, the run fails.  Without a separate icetop trigger, it is not possible to achieve 10 consecutive failed watchdogs due to buffer alone.

**Comment by dglo** [comment]
[tbendfelt on 2016-12-20 15:49:44]
I opened to reduce but not eliminate the buffer for both the hit channel and the data request channel.

-Dicecube.daq.stringhub.StringHubComponent.hitOutType=BLOCKING_1K
-Dicecube.daq.stringhub.StringHubComponent.dataOutType=BLOCKING_1K

Profiling on SPTS with scube showed that eliminating the buffer completely gave back some of the CPU gains while shrinking it to 1K was nearly indistinguishable from 128K as far as cpu utilization.

At 1K, a per-hub  LC hit rate as low as 5.12Hz is supported while ensuring that the trigger is fed new data within each 10 second watchdog period.

svn log -v -r 16404:16405 sps-cluster.cfg 
------------------------------------------------------------------------
[commit c80b6ed525485952e6f10390294d0b94eaffd63c] r16404 | bendfelt | 2016-12-20 15:40:54 -0600 (Tue, 20 Dec 2016) | 1 line
Changed paths:
   M /projects/config/trunk/sps-cluster.cfg

Issue 8014: Reduce hit channel buffering from 128K to 1K and data channel buffering from 8K to 1K.
------------------------------------------------------------------------
[commit c80b6ed525485952e6f10390294d0b94eaffd63c] r16405 | bendfelt | 2016-12-20 15:43:22 -0600 (Tue, 20 Dec 2016) | 1 line
Changed paths:
   M /projects/config/trunk/sps-cluster.cfg

Issue 8014: Removed extra quote.
------------------------------------------------------------------------
